File: Linguagem de Programação I/Extras/Beecrowd_Extra/1212.c
#include <stdio.h>
#include <string.h>

int main() {
    char a[11], b[11];
    while (1) {
        scanf("%s %s", a, b);
        if (a[0] == '0' && b[0] == '0' && a[1] == '\0' && b[1] == '\0') {
            break;
        }

        int carry = 0, c = 0;
        int len_a = strlen(a), len_b = strlen(b);
        while (len_a < len_b) {
            memmove(a + 1, a, len_a + 1);
            a[0] = '0';
            len_a++;
        }
        while (len_b < len_a) {
            memmove(b + 1, b, len_b + 1);
            b[0] = '0';
            len_b++;
        }

        for (int i = len_a - 1; i >= 0; i--) {
            if ((a[i] - '0' + b[i] - '0' > 9) || (a[i] - '0' + b[i] - '0' + c > 9)) {
                carry++;
                c = 1;
            } else {
                c = 0;
            }
        }

        if (carry == 0) {
            printf("No carry operation.\n");
        } else if (carry == 1) {
            printf("1 carry operation.\n");
        } else {
            printf("%d carry operations.\n", carry);
        }
    }
    return 0;
}

File: Linguagem de Programação I/Extras/Beecrowd_Extra/1212.php
<?php

while (true) {
    $input = explode(" ", trim(fgets(STDIN)));
    $a = strval($input[0]);
    $b = strval($input[1]);
    $carry = 0;
    $c = 0;
    if ($a == '0' && $b == '0') {
        break;
    }

    while (strlen($a) < strlen($b)) {
        $a = '0' . $a;
    }
    while (strlen($b) < strlen($a)) {
        $b = '0' . $b;
    }

    for ($i = strlen($a) - 1; $i >= 0; $i--) {
        if ((intval($a[$i]) + intval($b[$i]) > 9) || (intval($a[$i]) + intval($b[$i]) + $c > 9)) {
            $carry += 1;
            $c = 1;
        } else {
            $c = 0;
        }
    }

    if ($carry == 0) {
        echo "No carry operation.\n";
    } elseif ($carry == 1) {
        echo "1 carry operation.\n";
    } else {
        echo "$carry carry operations.\n";
    }
}

?>